drivers/sensor/bq34110/bq34110.c

```diff
@@ -0,0 +1,381 @@
+ *
+ * Registers have an address and 16-bit value
+ *
+ * @param bq34110 Private data for the driver
+ * @param reg_addr Register address to read
+ * @param val Place to put the value on success
+ * @return 0 if success, or negative error code from I2C_API
+ */
+static int bq34110_cmd_reg_read(const struct device *dev,
+				uint8_t reg_addr, int16_t *val)
+{
+	int ret;
+	uint8_t i2c_data[2];
+	struct bq34110_data *priv = dev->data;
+	const struct bq34110_config *cfg = dev->config;
+
+	ret = i2c_write_read(priv->i2c, cfg->i2c_addr, &reg_addr,
+			     sizeof(reg_addr), &i2c_data, 2);
+	if (ret < 0) {
+		LOG_ERR("Unable to read register");
+		return ret;
+	}
+
+	*val = ((i2c_data[1] << 8) | i2c_data[0]);
+
+	return 0;
+}
+
+/**
+ * @brief sensor value get
+ *
+ * @param dev BQ34110 device to access
+ * @param chan Channel number to read
+ * @param val Returns sensor value read on success
+ * @return 0 if successful
+ * @return -ENOTSUP for unsupported channels
+ */
+static int bq34110_channel_get(const struct device *dev,
+			       enum sensor_channel chan,
+			       struct sensor_value *val)
+{
+	float int_temp;
+	int32_t avg_power;
+	struct bq34110_data *priv = dev->data;
+
+	switch (chan) {
+	case SENSOR_CHAN_GAUGE_TEMP:
+		/* Convert the temperature in kelvin to deg celcius */
+		int_temp = (priv->internal_temperature * 0.1);
+		int_temp = int_temp - 273.15;
+		val->val1 = (int32_t)int_temp;
+		val->val2 = (int_temp - (int32_t)int_temp) * 1000000;
+		break;
+
+	case SENSOR_CHAN_GAUGE_VOLTAGE:
+		val->val1 = (priv->voltage / 1000);
+		val->val2 = ((priv->voltage % 1000) * 1000U);
+		break;
+
+	case SENSOR_CHAN_GAUGE_AVG_POWER:
+		avg_power = (priv->avg_power * 10);
+		val->val1 = (avg_power / 1000);
+		val->val2 = ((avg_power % 1000) * 1000U);
+		break;
+
+	case SENSOR_CHAN_GAUGE_AVG_CURRENT:
+		val->val1 = (priv->avg_current / 1000);
+		val->val2 = ((priv->avg_current % 1000) * 1000U);
+		break;
+
+	case SENSOR_CHAN_GAUGE_STATE_OF_HEALTH:
+		val->val1 = priv->state_of_health;
+		val->val2 = 0;
+		break;
+
+	case SENSOR_CHAN_GAUGE_STATE_OF_CHARGE:
+		val->val1 = priv->state_of_charge;
+		val->val2 = 0;
+		break;
+
+	case SENSOR_CHAN_GAUGE_TIME_TO_FULL:
+		/* Get time in minute */
+		val->val1 = priv->time_to_full;
+		val->val2 = 0;
+		break;
+
+	case SENSOR_CHAN_GAUGE_TIME_TO_EMPTY:
+		/* Get time in minute */
+		val->val1 = priv->time_to_empty;
+		val->val2 = 0;
+		break;
+
+	case SENSOR_CHAN_GAUGE_FULL_CHARGE_CAPACITY:
+		val->val1 = (priv->full_charge_capacity / 1000);
+		val->val2 = ((priv->full_charge_capacity % 1000) * 1000U);
+		break;
+
+	case SENSOR_CHAN_GAUGE_REMAINING_CHARGE_CAPACITY:
+		val->val1 = (priv->remaining_charge_capacity / 1000);
+		val->val2 =
+			((priv->remaining_charge_capacity % 1000) * 1000U);
+		break;
+
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static int bq34110_sample_fetch(const struct device *dev,
+				enum sensor_channel chan)
+{
+	struct bq34110_data *priv = dev->data;
+	struct {
```

Comment:
Does it make sense to change this to `static const struct`?

Reply:
`const` is not required here. since `dest` variable to used the read the gauge parameter's.

Reply:
Ack. Yeah, I missed that it was local.